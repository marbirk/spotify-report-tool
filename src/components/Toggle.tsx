import React from 'react'

interface ToggleProps {
    checked: boolean
    onChange: () => void
}

const Toggle = ({ checked, onChange }: ToggleProps) => (
    <span className="flex relative items-center pt-0 pr-1 pl-1">
        <input
            className="bg-gray-500 relative w-10 h-3 rounded-md cursor-pointer appearance-none outline-none dmcheck"
            type="checkbox"
            checked={checked}
            onChange={onChange}
            id="dmcheck"
        />
        <label htmlFor="dmcheck" />
    </span>
)

export default Toggle
