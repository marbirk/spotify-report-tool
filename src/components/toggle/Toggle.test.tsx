import React from 'react'
import { render } from '@testing-library/react'

import Toggle from './Toggle'

describe('Toggle', () => {
    test('renders correctly', () => {
        const { container, getByTestId } = render(
            <Toggle checked={true} onChange={() => {}} />
        )
        expect(getByTestId('toggle')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})
