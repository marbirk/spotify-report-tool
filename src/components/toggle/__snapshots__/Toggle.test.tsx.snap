// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly 1`] = `
<div>
  <span
    class="flex relative items-center pt-0 pr-1 pl-1"
  >
    <input
      checked=""
      class="bg-gray-500 relative w-10 h-3 rounded-md cursor-pointer appearance-none outline-none dmcheck"
      data-testid="toggle"
      id="dmcheck"
      type="checkbox"
    />
    <label
      for="dmcheck"
    />
  </span>
</div>
`;
